src/Datadog.Trace/Configuration/ConfigurationSourceExtensions.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Datadog.Trace.Configuration
+{
+    internal static class ConfigurationSourceExtensions
+    {
+        private const char Splitter = ';';
+
+        public static IEnumerable<string> GetStrings(this IConfigurationSource source, string key)
+        {
+            return source?.GetString(key)
+                ?.Split(new[] { Splitter }, StringSplitOptions.RemoveEmptyEntries)
+                ?? Enumerable.Empty<string>();
+        }
+
+        public static TEnum GetTypedValue<TEnum>(this IConfigurationSource source, string key)
+            where TEnum : struct, IConvertible
+        {
+            Enum.TryParse(source?.GetString(key) ?? "default", ignoreCase: true, out TEnum typedValue);
+            return typedValue;
+        }
+
+        public static HashSet<TEnum> GetTypedValues<TEnum>(this IConfigurationSource source, string key)
+            where TEnum : struct, IConvertible
```

Comment:
```suggestion
            where TEnum : Enum
```

More: https://stackoverflow.com/a/8086788/859275

Reply:
wow, they finally did it, thanks for the info!

Reply:
Unfortunately TryParse doesn't use it yet. I'll keep it as is - currently the solution seems more of using KISS.

Reply:
resolve